File: hotel_project/Frontend/src/services/tableService.js
import apiClient from '../api/apiClient';

/**
 * Table service — consumes generic REST APIs.
 */
const tableService = {
  getTables: async () => {
    const response = await apiClient.get('/tables');
    return response.data;
  },

  updateTableStatus: async (tableId, status) => {
    const response = await apiClient.put(`/tables/${tableId}`, { status });
    return response.data;
  },

  createTable: async (id, seats) => {
    const response = await apiClient.post('/tables', { id, seats });
    return response.data;
  },

  deleteTable: async (tableId) => {
    await apiClient.delete(`/tables/${tableId}`);
  },
};

export default tableService;

File: hotel_project/pos-backend/src/main/java/com/restaurent/pos_backend/service/TableService.java
package com.restaurent.pos_backend.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.restaurent.pos_backend.model.RestaurantTable;
import com.restaurent.pos_backend.repository.RestaurantTableRepository;

@Service
public class TableService {
	private final RestaurantTableRepository tableRepository;
	
	public TableService(RestaurantTableRepository tableRepository) {
		this.tableRepository = tableRepository;
	}
	
	public List<RestaurantTable> getAllTables(){
		return tableRepository.findAll();
	}
	
	public RestaurantTable updateTableStatus(String tableId, String newStatus) {
		RestaurantTable table = tableRepository.findById(tableId)
				                .orElseThrow(() -> new RuntimeException("Table not found!"));
				
				table.setStatus(newStatus);
				return tableRepository.save(table);
	}

	public RestaurantTable createTable(String id, Integer seats) {
		if (id == null || id.isBlank()) {
			throw new RuntimeException("Table ID is required!");
		}
		if (tableRepository.existsById(id)) {
			throw new RuntimeException("Table " + id + " already exists!");
		}

		RestaurantTable table = new RestaurantTable();
		table.setId(id);
		table.setSeats(seats != null ? seats : 2);
		table.setStatus("empty");
		return tableRepository.save(table);
	}

	public void deleteTable(String tableId) {
		RestaurantTable table = tableRepository.findById(tableId)
				.orElseThrow(() -> new RuntimeException("Table not found!"));

		if (!"empty".equalsIgnoreCase(table.getStatus())) {
			throw new RuntimeException("Cannot delete table " + tableId + " — it is not empty!");
		}

		tableRepository.deleteById(tableId);
	}
}
